DatadogSessionReplay/Sources/DatadogSessionReplay/Utilities/Cache.swift

```diff
@@ -0,0 +1,108 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+internal final class Cache<Key: Hashable, Value> {
```

Comment:
Implementation lat wraps swift types in NSObjects so we can use NSCache. It's quite performant. Used this approach a few times in the past.